### PR TITLE
Add ci configs/1

### PR DESCRIPTION
**Pull request by zahraaalizadeh**
This change:
- Fixes a few typos
- Adds a basic CI configuration to run `format` task on the subproject
- Adds a documentation about how to use 'act' to run github workflows locally.
- Updates the README file with how to include github workflows when starting  a new project.

issue: https://github.com/ackama/django-template/issues/12

### CODE REVIEW

File: template/docs/reference/entrypoint.md
Comment (a-musing-moose) [inline template/docs/reference/entrypoint.md:1]: **chore:** This should be a separate PR

This change, whilst useful, is unrelated to Continuous Integration and so should really be in PR of it's own.

File: template/.github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  static-analysis-check:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.12.2'
```
Comment (a-musing-moose) [inline template/.github/workflows/ci.yml:15]: **suggestion:** Limit the version specification to `major.minor` only

There really should not be any breaking changes between `patch` releases of Python. Given that, specifying the version down to the `patch` level means we are likely to need to update it more often than is really necessary. I'd instead change this to just `3.12`.‏

File: template/.github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        cache: 'poetry'
+
+    - name: Install dependencies
+      run: |
+        pip install invoke
+        pip install poetry
```
Comment (a-musing-moose) [inline template/.github/workflows/ci.yml:21]: **nitpick (non-blocking):** You could install both these package with a single call to pip:

```shell
pip install invoke poetry
```

File: template/docs/tutorials/your-first-day.md [hunk 3]
```diff
@@ -216,9 +216,26 @@ You can view your change once deployed at `https://<hostname>/humans.txt`
...
+To simulate the execution of GitHub Actions workflows on your local machine, use the following command:
+
+```shell
+act push --container-architecture linux/amd64
+```
+
+The `push` parameter triggers the workflow associated with the push event. You can replace push with any other GitHub event that your workflow listens to, such as `pull_request`.
+
+The `--container-architecture` option specifies the architecture of the container in which the workflow will run. linux/amd64 is commonly used, but you may need to adjust this based on your project's requirements or the architecture of your machine.
+
```
Comment (a-musing-moose) [inline template/docs/tutorials/your-first-day.md]: **issue:** This should not be part of someones first day

`act` is a really useful tool for testing github actions. I've used it myself a number of time because no matter what I do I never seem to be able to get a CI config to work first time!

That said, this document is about a developers _first day_ on a project. It is a tutorial aimed at guiding a developer through the steps needed to get the project up and running and making their first contribution. Working on CI pipelines outside of simply triggering them by opening a PR is well outside the scope of a developers first day.

The content is still valuable but I would suggest moving it to it's own tutorial or possible a how-to.

- Tutorials: Step by step guide where the reader is expected to follow the steps verbatim. Assumes little to no familiarity with the system.
- How-To: A more general guide as to help direct the reader's work. Can assume a greater degree of familiarity with the system

For more information of the differences see [The difference between a tutorial and how-to guide](https://diataxis.fr/tutorials-how-to/#). Take a look around that site as well, it details a well thought through approach to documentation and is worth learning about.

File: README.md [hunk 4]
```diff
@@ -25,9 +25,19 @@ This will ensure that the `django-admin` command is available in your shell. Fro
...
 
+> [!NOTE]
+> When starting a new Django project using a custom template, some directories and files
+> that begin with a dot (.) are not automatically included. This exclusion typically affects
+> important folders like `.github`, which is commonly used for GitHub Actions CI/CD workflows.
+> However, from Django 4.0, you can ensure that directories like `.github` are properly
+> copied to your new project. You can achieve this by using the `--exclude` option during
+> project initialization. If you omit this option, directories beginning with a dot, including
+> `.github/workflows`, will be left out.
+
```
Comment (a-musing-moose) [inline README.md:39]: **question:** ‏How does adding an exclusion mean that extra things get included?

I think I understand the problem, that Django's `startproject` command will ignore dot folders and files by default. But I am not sure I understand how adding a `--exclude` flag to make sure the `.venv` is ignored helps?

Aside: You shouldn't have a .venv in the template directory. So the exclude should not be necessary.

File: template/docs/tutorials/your-first-day.md
Comment (a-musing-moose) [inline template/docs/tutorials/your-first-day.md:1]: **issue:** ‏As before, using `act` to run github actions locally should not be part of your first day. A developer should be using `invoke` locally to run these task.

**Review by a-musing-moose — CHANGES_REQUESTED** [review]
This is really good work, but I think it needs to be split up into 2 PR and a little re-arrangement of the documentation is needed.

File: template/docs/reference/entrypoint.md
Comment (zahraaalizadeh) [reply template/docs/reference/entrypoint.md:1]: Opened https://github.com/ackama/django-template/pull/42 to address this.

File: README.md (hunk 4 above)
Comment (zahraaalizadeh) [reply README.md:39]: Thank you @a-musing-moose for the review. That's a good question. 

I agree that using `--exclude .venv` here seems unusual. The issue with copying the GitHub workflows lies in Django's default behaviour, which excludes all folders starting with a `.`. I've considered a few options to address this:

- Manual Copy: We could add notes to the README documentation and have users manually copy the GitHub actions from the template to their new project.
- Script Automation: Another option is to create scripts that run the `django-admin startproject` command followed by the `cp` command to copy the necessary files.
- Custom Exclude: Alternatively, passing `--exclude` here overrides the default behaviour, excluding only the directories specified. For example, using `--exclude <FICTIONAL DIRECTORY>` would prevent Django from excluding the `.github` folder. Currently, there are no folders in the template that we want to exclude, so this approach may seem unnecessary. However it actually lets the github workflows to be copied, and I believe it still offers a more coherent solution compared to the previous options.

I'm open to the suggestion though.

File: template/.github/workflows/ci.yml (hunk 2 above)
Comment (zahraaalizadeh) [reply template/.github/workflows/ci.yml]: Updated in https://github.com/ackama/django-template/commit/c31cf55160c355420fa8a874e1795922c1d1ccd6

File: template/.github/workflows/ci.yml (hunk 1 above)
Comment (zahraaalizadeh) [reply template/.github/workflows/ci.yml]: updated in https://github.com/ackama/django-template/commit/c31cf55160c355420fa8a874e1795922c1d1ccd6

File: template/docs/tutorials/your-first-day.md (hunk 3 above)
Comment (zahraaalizadeh) [reply template/docs/tutorials/your-first-day.md]: [TIL] Added a new how-to doc https://github.com/ackama/django-template/commit/bf09b91e1ff0a76efc76667b4756da8cf72aca66

File: template/docs/tutorials/your-first-day.md
Comment (zahraaalizadeh) [reply template/docs/tutorials/your-first-day.md:1]: Added a new how-to doc https://github.com/ackama/django-template/commit/bf09b91e1ff0a76efc76667b4756da8cf72aca66

File: README.md (hunk 4 above)
Comment (zahraaalizadeh) [reply README.md]: I updated the README doc to clarify this a bit more: [ccd6de7](https://github.com/ackama/django-template/pull/41/commits/ccd6de7dd183ad22e431210dc0e723549010f3d8)

File: README.md [hunk 5]
```diff
@@ -25,18 +25,16 @@ This will ensure that the `django-admin` command is available in your shell. Fro
...
-> copied to your new project. You can achieve this by using the `--exclude` option during
-> project initialization. If you omit this option, directories beginning with a dot, including
-> `.github/workflows`, will be left out.
-
+>When initiating a Django project with a custom template, be aware that directories starting with
+>a dot (e.g., `.github` for GitHub Actions workflows) are not included by default. A workaround
+>from Django 4.0 onwards involves using the `--exclude` option with the `startproject` command.
+>Oddly, specifying `--exclude` with a non-existent directory can allow these dot-prefixed
+>directories to be copied. This trick can ensure that essential configurations like `.github` are
+>included in your project setup.
```
Comment (a-musing-moose) [inline README.md]: **thought (non-blocking):** ‏I think we are rapidly approaching the point at which we need our own wrapper script for the generating a new project. This would allow us to do extra things like copy across the `.github` folder without resorting quirks, and also automatically add variants when we get some.

Not something we need right now though.

**Review by a-musing-moose — APPROVED** [review]
Awesome work. Thanks Zahra

File: README.md (hunk 5 above)
Comment (zahraaalizadeh) [reply README.md]: I agree Jon 👍🏻
